src/renderer/corporate/member/corporate-add-member-container.jsx

```diff
@@ -128,22 +216,35 @@ class CorporateAddMemberContainer extends PureComponent {
 }
 
 const mapStateToProps = (state, props) => {
-	const parentIdentity = identitySelectors.selectCorporateProfile(state, {
-		identityId: props.match.params.parentId
+	const identity = identitySelectors.selectIdentity(state);
+	let { parentId } = props.match.params;
+	parentId = parentId || identity.id;
+
+	const parentProfile = identitySelectors.selectCorporateProfile(state, {
+		identityId: parentId
 	});
 
+	if (!parentProfile || parentProfile.identity.type !== 'corporate') {
+		throw new Error(`Invalid parent identity, requires 'corporate' type`);
+	}
+
 	return {
-		parentIdentity,
-		basicAttributeTypes: identitySelectors.selectBasicCorporateAttributeTypes(state),
-		basicIdentity: identitySelectors.selectCorporateProfile(state, {
-			identityId: props.match.params.identityId
-		}),
+		parentId,
+		parentProfile,
+		identity,
+		individualAttributeTypes: identitySelectors.selectMemberIndividualAttributeTypes(state),
+		corporateAttributeTypes: identitySelectors.selectMemberCorporateAttributeTypes(state),
 		walletType: appSelectors.selectWalletType(state),
+		countries: identitySelectors.selectCountries(state),
 		jurisdictions: identitySelectors.selectCorporateJurisdictions(state),
 		entityTypes: identitySelectors.selectCorporateLegalEntityTypes(state),
-		positions: identitySelectors.selectPositionsForCompanyType(state, {
-			companyType: parentIdentity.profile.entityType
-		})
+		availablePositions: identitySelectors.selectPositionsForCompanyType(state, {
+			companyType: parentProfile.entityType
+		}),
+		companies: [
+			parentProfile,
+			...parentProfile.members.filter(m => m.identity.type === 'corporate')
```

Comment:
probably should be in the profile selector

Reply:
What do you mean?
Have a new corporate profile attribute called companies, instead of aggregating them here?

Reply:
yes, that what I meant, that way when we have memoization we will not need to filter.
But nvm for now if you do it only here

Reply:
Agree, we should strive to use memoization everywhere.
Improvement on 49e4583